### PR TITLE
add proxy match on version for telemetry v2 filters

### PR DESCRIPTION
**Pull request by richardwxn**

### CODE REVIEW

**Comment by richardwxn** [comment]
@mandarjog I am still testing locally but PTAL of overall idea

**Comment by mandarjog** [comment]
I think this looks ok, though the version specific values should not say “*”
Please ensure that you have both positive and negative tests.

File: istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -138,3 +138,7 @@ telemetry:
     # For Null VM case now. If enabled, will set disableMixerHttpReports to true and not define mixerReportServer
     # also enable metadata exchange and stats filter.
     enabled: false
+
+    proxyMatch:
+      # regular expression to be used to select proxies to apply telemetry v2 envoy filters. e.g. "1.5*"
+      proxyVersion: "*"
```
Comment (howardjohn) [inline istio-control/istio-discovery/values.yaml]: Shouldn't this be `.*`?

And above would need to be `1\.5.*`?

**Comment by mandarjog** [comment]
@richardwxn please ensure that we emit a proper error, afaik we are silently ignoring the error and then it means “match anything”

File: istio-control/istio-discovery/values.yaml (hunk 1 above)
Comment (richardwxn) [reply istio-control/istio-discovery/values.yaml]: yep it should be, I forgot to update

**Comment by richardwxn** [comment]
@mandarjog Can you give an example of the case when error is silenced. Actually if you provide a bad type of the proxyMatch field it would just fail validation and does not apply. But if you supply a regular expression A(assume the format is correct) although it is not you intend to do it would just match A but would not match anything

**Pull request closed without merging**